server/cluster/cluster.go

```diff
@@ -532,6 +532,17 @@ func (c *Cluster) GetTable(ctx context.Context, schemaName, tableName string) (*
 	return nil, false, nil
 }
 
+func (c *Cluster) GetShardByID(id uint32) (*Shard, error) {
```

Comment:
Maybe we shuold add comments to tell the caller the returned Shard is immutable.